compiler/rustc_mir_dataflow/src/drop_flag_effects.rs

```diff
@@ -155,16 +155,15 @@ where
     }
 }
 
-/// Calls `handle_inactive_variant` for each descendant move path of `enum_place` that contains a
-/// `Downcast` to a variant besides the `active_variant`.
-///
-/// NOTE: If there are no move paths corresponding to an inactive variant,
-/// `handle_inactive_variant` will not be called for that variant.
-pub(crate) fn on_all_inactive_variants<'tcx>(
+/// Handles each variant that corresponds to one of the child move paths of `enum_place`. If the
+/// variant is `active_variant`, `handle_active_variant` will be called (if provided). Otherwise,
+/// `handle_inactive_variant` will be called.
+pub(crate) fn on_all_variants<'tcx>(
     move_data: &MoveData<'tcx>,
     enum_place: mir::Place<'tcx>,
     active_variant: VariantIdx,
     mut handle_inactive_variant: impl FnMut(MovePathIndex),
+    mut handle_active_variant: Option<impl FnMut(MovePathIndex)>,
```

Comment:
I wonder if it will be better, perf-wise to make this not take an option but just use a no-op for callers that do not want to handle it.